01_loaddata.R

```diff
@@ -60,46 +59,22 @@ birds_sf<- birds_sf|>
    dplyr::mutate(bearing = bearing(c(location.long_prior,location.lat_prior), c(Longitude, Latitude)),
                  speed_kmh = round((dist_km /timediff_hrs),1))
 
-timediff_hrs <- c()
+# 4) Calculate Geoid Height from Ellipsoid Height
 
-for(i in 1:nrow(birds_sf)) {
-  if(i == 1){
-    timediff_hrs[i] <- 0
-  }
-  else{
-    d <- as.numeric(difftime(birds_sf[i,]$Date, birds_sf[i-1,]$Date, units = "hours")) # time difference for date fields
-    t <- as.numeric(difftime(birds_sf[i,]$Time, birds_sf[i-1,]$Time, units = "hours")) # time difference for time fields (need to account for if dates are not equal)
-    if(t < 0){
-      timediff_hrs[i] <- d + (24 + t) # adjusts time difference where dates differ and i time is a lower value than i-1 time
-    }
-    else{
-      timediff_hrs[i] <- d + t # add the total time difference for date and time fields
-    }
-  }
-}
+EGM2008_1 <- rast(path("01_inputs", "us_nga_egm2008_1.tif")) # The Earth Gravitational Geoid Model 2008 1'
 
-birds_sf$timediff_hrs <- timediff_hrs
+geoid_height <- extract(EGM2008_1, vect(birds_sf)) # tested these calculated geoid values against the geoid height calculator and the values matched
```

Comment:
I get a warning message when I use this line '''Warning message:
[extract] transforming vector data to the CRS of the raster '''. 
Might want to consider reprojecting the bird data to the CRS that matches the base grid then convert back to WGS84?